### PR TITLE
KUDO Kafka Platform Service Docs

### PR DESCRIPTION
**Pull request by kaiwalyajoshi**
## Jira Ticket

[D2IQ-74568](https://jira.d2iq.com/browse/D2IQ-74568)

## Checklist
- [ ] Change all affected versions, if applicable (e.g. 1.13, 2.0, 2.1).
- [ ] Test all commands and procedures, if applicable.
- [ ] Create your PR against `main`. 
- [ ] Update all links if you are moving a page.
- [ ] Add release date to Release Notes page in the following format: <Package> was released on <Day>, <Month> <Year> Example: `Mesosphere® DC/OS™ 2.1.0 was released on 9, June 2020`

See the [contribution guidelines](https://wiki.d2iq.com/display/ENG/Contributing+to+Docs) for more information.

### CODE REVIEW

**Comment by kaiwalyajoshi** [comment]
Depends on #3530 and should be merged after that PR is merged.

**Comment by kaiwalyajoshi** [comment]
Docs Preview Page found here: http://docs-d2iq-com-pr-3532.s3-website-us-west-2.amazonaws.com/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/

File: pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+Kommander Catalog adds integration for [KUDO Kafka Operator](https://github.com/mesosphere/kudo-kafka-operator/), which simplifies day 2 operations of [Apache Kafka](https://kafka.apache.org/). 
+
+#include /dkp/kommander/1.3/include/kudo-intro.tmpl
+
+It is **strongly recommended** to view the [KUDO Kafka Documentation](https://kudo.dev/docs/runbooks/kafka/install.html) which covers the KUDO Kafka Operator in-depth. This document covers the integration aspects of KUDO Kafka Operator with D2iQ Kommander.
+
+### Kommander Catalog
+
+KUDO Kafka is located in the Kommander Catalog.   To access the catalog:
```
Comment (rishabh96b) [inline pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md:22]:
```suggestion
KUDO Kafka is located in the Kommander Catalog. To access the catalog: 
```

File: pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+- Kommander will proceed to install `kudo-controller-manager` in the `kudo-system` namespace if it does not already exist. This is the equivalent of issuing `kubectl kudo init` manually on the CLI.
+- KUDO will then install Kafka in the Project namespace created using Kommander.
+- The deployment progression can be viewed by looking at the `deploy` plan.
+
+### KUDO Kafka Service Status
+
+The Operator Plan Status provides an overview of the service.
+
+At this point it is useful to have an understanding of [KUDO Operator Plans](https://kudo.dev/docs/what-is-kudo.html#operator-plans) and the [KUDO CLI](https://kudo.dev/docs/cli/installation.html) should be installed.
```
Comment (rishabh96b) [inline pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md:58]:
```suggestion
At this point, it is useful to have an understanding of [KUDO Operator Plans](https://kudo.dev/docs/what-is-kudo.html#operator-plans) and the [KUDO CLI](https://kudo.dev/docs/cli/installation.html) should be installed.
```

File: pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+KUDO Kafka comes integrated with [Cruise Control](https://github.com/linkedin/cruise-control)
+
+Cruise Control is a tool to fully automate the dynamic workload rebalance and self-healing of a Kafka cluster. It provides great value to Kafka users by simplifying the operation of Kafka clusters.
+
+Cruise Control integration is disabled by default, details on how to enable it are found in the [Kafka Operator docs](https://github.com/kudobuilder/operators/blob/master/repository/kafka/docs/latest/cruise-control.md)
+
+### External Access
+
+For Producers and Consumers living outside the Kubernetes cluster, KUDO supports Loadbalancer and NodePort approaches, the Loadbalancer approach is the recommended one.
```
Comment (rishabh96b) [inline pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md:188]:
```suggestion
For Producers and Consumers living outside the Kubernetes cluster, KUDO supports LoadBalancer and NodePort approaches, the LoadBalancer approach is the recommended one.
```

**Review by rishabh96b — CHANGES_REQUESTED** [review]

File: pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+```
+Wait for the deployment plan to `COMPLETE` as shown in the Kafka broker counts example.
+
+
+### Upgrades
+
+KUDO Kafka versions can be upgraded using the KUDO CLI.
+
+**Example** Upgrade KUDO Kafka from `v2.5.0-1.3.1` to `v2.5.1-1.3.3`:
+When upgrading, you should understand the mapping between Kafka versions and operator versions. For more information, see the table at the end of the [Kafka operator](https://github.com/mesosphere/kudo-kafka-operator/tree/master/docs)
```
Comment (rishabh96b) [inline pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md:135]:
```suggestion
When upgrading, you should understand the mapping between Kafka versions and operator versions. For more information, see the table at the end of the [Kafka operator](https://github.com/mesosphere/kudo-kafka-operator/tree/master/docs) repository.
```

File: pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md (hunk 1 above)
Comment (rishabh96b) [inline pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md:22]:
```suggestion
KUDO Kafka service is located in the Kommander Catalog.   To access the catalog: 
```

File: pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+
+```
+kubectl kudo upgrade kafka --instance kafka5 --operator-version 1.3.3
```
Comment (rishabh96b) [inline pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md:138]:
```suggestion
kubectl kudo upgrade kafka --instance kafka --operator-version 1.3.3
```

File: pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+![KUDO Kafka Monitoring](/dkp/kommander/1.3/img/platform-services-kafka-monitoring.png)
+
+
+### Kafka MirrorMaker
+
+KUDO Kafka comes integrated with [MirrorMaker](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=27846330)
+
+MirrorMaker is a tool to mirror a source Kafka cluster into a target (mirror) Kafka cluster. This tool uses a Kafka consumer and a Kafka producer. The consumer consumes messages from the source cluster. The producer re-publishes those messages to the target cluster.
+
+MirrorMaker integration is disabled by default, details on how to enable it are found in the [Kafka Operator docs](https://github.com/kudobuilder/operators/blob/master/repository/kafka/docs/latest/mirrormaker.md)
```
Comment (rishabh96b) [inline pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md:169]:
```suggestion
MirrorMaker integration is disabled by default, details on how to enable it are found in the [Kafka Operator docs](https://github.com/kudobuilder/operators/blob/master/repository/kafka/docs/latest/mirrormaker.md).
```

File: pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+
+### Kafka Connect
+
+KUDO Kafka comes integrated with [Kafka Connect](https://kafka.apache.org/documentation/#connect)
```
Comment (rishabh96b) [inline pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md:173]:
```suggestion
KUDO Kafka comes integrated with [Kafka Connect](https://kafka.apache.org/documentation/#connect).
```

File: pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+
+Kafka Connect is a tool for scalably and reliably streaming data between Apache Kafka and other systems. It provides a REST API to configure and interact connectors.
+
+Kafka Connect integration is disabled by default, details on how to enable it are found in the [Kafka Operator docs](https://github.com/kudobuilder/operators/blob/master/repository/kafka/docs/latest/kafka-connect.md)
```
Comment (rishabh96b) [inline pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md:177]:
```suggestion
Kafka Connect integration is disabled by default, details on how to enable it are found in the [Kafka Operator docs](https://github.com/kudobuilder/operators/blob/master/repository/kafka/docs/latest/kafka-connect.md).
```

File: pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+### Cruise Control
+
+KUDO Kafka comes integrated with [Cruise Control](https://github.com/linkedin/cruise-control)
```
Comment (rishabh96b) [inline pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md:180]:
```suggestion
KUDO Kafka comes integrated with [Cruise Control](https://github.com/linkedin/cruise-control).
```

File: pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md [hunk 10]
```diff
@@ -0,0 +1,212 @@
...
+
+Cruise Control is a tool to fully automate the dynamic workload rebalance and self-healing of a Kafka cluster. It provides great value to Kafka users by simplifying the operation of Kafka clusters.
+
+Cruise Control integration is disabled by default, details on how to enable it are found in the [Kafka Operator docs](https://github.com/kudobuilder/operators/blob/master/repository/kafka/docs/latest/cruise-control.md)
```
Comment (rishabh96b) [inline pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md:184]:
```suggestion
Cruise Control integration is disabled by default, details on how to enable it are found in the [Kafka Operator docs](https://github.com/kudobuilder/operators/blob/master/repository/kafka/docs/latest/cruise-control.md).
```

**Review by rishabh96b — CHANGES_REQUESTED** [review]

File: pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md [hunk 11]
```diff
@@ -0,0 +1,212 @@
...
+The diagnostics data contains the following:
+
+KUDO Environment
+- Installed Manager and its logs.
+- Service account and services.
+
+Data for the specified Operator
+- The Operator, OperatorVersion and Instance resources.
+- Deployed resources from the operator.
+- Logs from deployed pods
```
Comment (rishabh96b) [inline pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md:207]:
```suggestion
- Logs from the deployed pods.
```

File: pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md (hunk 1 above)
Comment (kaiwalyajoshi) [reply pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md:22]: Not to bring up a holy war here but this is what we were taught in Grade 9 (I think) typesetting class. ;)

https://www.thesaurus.com/e/writing/is-there-1-space-or-2-after-a-period/

File: pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md (hunk 5 above)
Comment (kaiwalyajoshi) [reply pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md:138]: Thanks for catching this!

**Review by rishabh96b — APPROVED** [review]
LGTM :rocket:

File: pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md [hunk 12]
```diff
@@ -0,0 +1,212 @@
...
+
+The `ZOOKEEPER_PATH` parameter configures the zNode path within the Zookeeper Cluster to use.
+- When unspecified the zNode defaults to the id of the current Kafka instance being launched.
+- When specified a mandatory leading slash is needed i.e `/kafka`
+
+### Installation
+
+From the [Project Catalog](/dkp/kommander/1.3/projects/platform-services/) select the desired version of Kafka and click Deploy.
+
+Below is an example of what should appear in the Kommander UI.  The dialog is populated with sensible defaults:
```
Comment (cneth) [inline pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md:41]: I just tried to install Kafka 2.5.1-1.  The 'sensible default' :-) for the zookeeper URI is this string:

`zookeeper-zookeeper-0.zookeeper-hs:2181,zookeeper-zookeeper-1.zookeeper-hs:2181,zookeeper-zookeeper-2.zookeeper-hs:2181`

This is what led to a lot of my confusion, in particular the reference to the zookeeper-hs namespace...  Probably not a docs issue, tho.  I can confirm that using <zk-install-name>-cs.<namespace>:2181 works here.

**Review by cneth — APPROVED** [review]

**Review by russurquhart1 — APPROVED** [review]

File: pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md (hunk 12 above)
Comment (kaiwalyajoshi) [reply pages/dkp/kommander/1.3/projects/platform-services/platform-services-catalog/kudo-kafka/index.md:41]: @cneth yeah, we're fixing that in upstream KUDO :)

**Pull request merged**